packages/react-accordion/src/components/AccordionHeader/useAccordionHeader.ts

```diff
@@ -59,53 +44,30 @@ export const useAccordionHeader = (
   const id = useId('accordion-header-', props.id);
   const panel = useDescendants(accordionItemDescendantContext)[1] as AccordionItemDescendant | undefined;
   const innerRef = React.useRef<HTMLElement>(null);
-  const state = mergeProps(
-    {
-      ref: useMergedRefs(ref, innerRef),
-      size: 'medium' as AccordionHeaderSize,
-      inline: false,
-      expandIcon: {
-        as: DefaultExpandIcon,
-        'aria-hidden': true,
-      },
-      button: {
-        id,
-        onClick: onAccordionHeaderClick,
-        'aria-disabled': disabled,
-        'aria-controls': panel?.id,
-      },
-      as: 'div',
-      role: 'heading',
-      expandIconPosition: 'start',
-      context: {
-        disabled: false,
-        open: false,
-        size: 'medium',
-        expandIconPosition: 'start',
-      },
+  const initialState = {
+    ...props,
+    ref: useMergedRefs(ref, innerRef),
+    size: props.size ?? size ?? 'medium',
+    inline: props.inline ?? inline ?? false,
+    role: props.role ?? 'heading',
+    expandIconPosition: props.expandIconPosition ?? expandIconPosition ?? 'start',
```

Comment:
```suggestion
    size: size || 'medium',
    inline: inline || false,
    role: 'heading',
    expandIconPosition: expandIconPosition || 'start',
    ...props,

    ref: useMergedRefs(ref, innerRef),
```

Should it be just like this?

Reply:
No, in the case user actually passes size as undefined, which is an actual expected value, things will break if I do that

Reply:
Should I avoid nullish coalescing?!